src/rules/color-named/__tests__/index.js

```diff
@@ -286,3 +286,71 @@ testRule(rule, {
     column: 22,
   } ],
 })
+
+testRule(rule, {
+  ruleName,
+  config: [ "never", { ignore: ["inside-function"] } ],
+
+  accept: [ {
+    code: "a { color: #000; }",
+  }, {
+    code: "a { color: gray(70%); }",
+  }, {
+    code: "a { background-image: url(red); }",
+  }, {
+    code: "a { color: map-get($colour, blue(60%)); }",
+  }, {
+    code: "a { color: map-get($colour, blue); }",
+  } ],
+
+  reject: [ {
+    code: "a { color: map-get($colour, blue), rebeccapurple; }",
+    message: messages.rejected("rebeccapurple"),
+    line: 1,
+    column: 36,
+  }, {
+    code: "a { color: rebeccapurple, map-get($colour, blue); }",
+    message: messages.rejected("rebeccapurple"),
+    line: 1,
+    column: 12,
+  }, {
+    code: "a { color: red; }",
+    message: messages.rejected("red"),
+    line: 1,
+    column: 12,
+  } ],
+})
+
+testRule(rule, {
+  ruleName,
+  config: [ "always-where-possible", { ignore: ["inside-function"] } ],
+
+  accept: [ {
+    code: "a { color: black; }",
+  }, {
+    code: "a { color: rgb(0, 0, 0); }",
```

Comment:
These aren't *inside* a function, they *are* a function and, as such, should fail.